Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -0,0 +1,2727 @@
+        fullExtentEndTimeLabel.fontSize = fullExtentLabelFont.pointSize
+        layer.addSublayer(fullExtentEndTimeLabel)
+        
+        // Add the lower value label
+        currentExtentStartTimeLabel.isHidden = !(labelMode == .thumbs && isSliderVisible)
+        currentExtentStartTimeLabel.foregroundColor = currentExtentLabelColor.cgColor
+        currentExtentStartTimeLabel.alignmentMode = kCAAlignmentCenter
+        currentExtentStartTimeLabel.frame = CGRect.zero
+        currentExtentStartTimeLabel.contentsScale = UIScreen.main.scale
+        currentExtentStartTimeLabel.font = currentExtentLabelFont as CFTypeRef
+        currentExtentStartTimeLabel.fontSize = currentExtentLabelFont.pointSize
+        layer.addSublayer(currentExtentStartTimeLabel)
+        
+        // Add the upper value label
+        currentExtentEndTimeLabel.isHidden = !(labelMode == .thumbs && isSliderVisible)
+        currentExtentEndTimeLabel.foregroundColor = currentExtentLabelColor.cgColor
+        currentExtentEndTimeLabel.alignmentMode = kCAAlignmentCenter
+        currentExtentEndTimeLabel.frame = CGRect.zero
+        currentExtentEndTimeLabel.contentsScale = UIScreen.main.scale
+        currentExtentEndTimeLabel.font = currentExtentLabelFont as CFTypeRef
+        currentExtentEndTimeLabel.fontSize = currentExtentLabelFont.pointSize
+        layer.addSublayer(currentExtentEndTimeLabel)
+        
+        // Create the images
+        let bundle = Bundle(for: type(of: self))
+        playImage = UIImage(named: "Play", in: bundle, compatibleWith: nil)
+        pauseImage = UIImage(named: "Pause", in: bundle, compatibleWith: nil)
+        forwardImage = UIImage(named: "Forward", in: bundle, compatibleWith: nil)
+        backImage = UIImage(named: "Back", in: bundle, compatibleWith: nil)
+        
+        // Setup Play/Pause button
+        playPauseButton.setImage(playImage, for: .normal)
+        playPauseButton.setImage(pauseImage, for: .selected)
+        playPauseButton.tintColor = playbackButtonsFillColor
+        playPauseButton.addTarget(self, action: #selector(TimeSlider.playPauseAction(_:)), for: .touchUpInside)
+        playPauseButton.showsTouchWhenHighlighted = true
+        addSubview(playPauseButton)
+        
+        // Setup forward button
+        forwardButton.setImage(forwardImage, for: .normal)
+        forwardButton.tintColor = playbackButtonsFillColor
+        forwardButton.addTarget(self, action: #selector(TimeSlider.forwardAction(_:)), for: .touchUpInside)
+        forwardButton.showsTouchWhenHighlighted = true
+        addSubview(forwardButton)
+        
+        // Setup back button
+        backButton.setImage(backImage, for: .normal)
+        backButton.tintColor = playbackButtonsFillColor
+        backButton.addTarget(self, action: #selector(TimeSlider.backAction(_:)), for: .touchUpInside)
+        backButton.showsTouchWhenHighlighted = true
+        addSubview(backButton)
+        
+        // Refresh
+        setNeedsLayout()
```

Comment:
While not harmful, I don't think this is needed. The system should request that the view layout its subviews before it is displayed.

Reply:
Removed